zeroeq/receiver.cpp

```diff
@@ -180,6 +177,22 @@ Receiver::~Receiver()
     _impl->remove(this);
 }
 
+Receiver::Receiver(Receiver&& from)
+    : _impl(from._impl)
+{
+    from._impl.reset();
+}
+
+Receiver& Receiver::operator=(Receiver&& from)
```

Comment:
But you are talking about doing it in the header, right? What I'm saying is that it can be done in the cpp file instead, as shown in the link above. You still need to forward-declare it, but it saves writing the implementation.